lib/Conversion/TritonGPUToLLVM/MemoryOpToLLVM.cpp

```diff
@@ -26,10 +26,9 @@ void lowerDistributedToShared(Operation *op, Value src, Value dst,
   auto dstShapePerCTA = triton::gpu::getShapePerCTA(dstTy);
   auto srcLayout = srcTy.getEncoding();
   auto outOrd = mlir::cast<SharedEncodingAttr>(dstTy.getEncoding()).getOrder();
-  assert(srcTy.getShape().size() == 2 ||
-         (srcTy.getShape().size() <= 3 && outOrd[2] == 0) &&
+  assert(srcTy.getShape().size() <= 2 ||
+         (srcTy.getShape().size() == 3 && outOrd[2] == 0) &&
```

Comment:
I can comment on code below but line 36 I believe the strides should come from the shared memory object for the StoreOp

Reply:
Ugh, that makes sense. Let me take another look at this.